CDS/scripts/atom_image.pl

```diff
@@ -70,12 +73,31 @@ sub is_imageurl_valid {
 
 my $capi = decode_json($response->content);
 
-unless (is_imageurl_valid($capi->{'response'}->{'media'}->{'data'}->{'media'}->{'trailImage'}->{'master'}->{'file'})) {
-	exit(1);
+if(defined $ENV{'output-smaller-than'}){
+  my %image_urls;
+  my $image_place = 0;
+  while($image_place < 20) {
+    if (defined $capi->{'response'}->{'media'}->{'data'}->{'media'}->{'trailImage'}->{'assets'}[$image_place]->{'file'}) {
```

Comment:
also, if I read the code correctly (translated to simplified version above) then it will get the _first_ match of an image which is smaller than the requested sizes, regardless of whether there is a larger one that is still within the sizes.

I would therefore suggest sorting the list before iterating https://perldoc.perl.org/functions/sort.html:

```
my $assets = $capi->{'response'}->{'media'}->{'data'}->{'media'}->{'trailImage'}->{'assets'};
my $match = undef;
my @sorted_assets = sort { $a->{'dimensions'}->{'height'} <=> $b->{'dimensions'}->{'height'} } @{$assets};

foreach(@sorted_assets){
```

(double-check the code as always though, not sure whether that will give ascending or descending!)

Reply:
This has been done.